packages/devtools_app/test/memory/allocation_tracing/allocation_profile_tracing_view_test.dart

```diff
@@ -0,0 +1,221 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:devtools_app/devtools_app.dart';
+import 'package:devtools_app/src/config_specific/import_export/import_export.dart';
+import 'package:devtools_app/src/screens/memory/memory_heap_tree_view.dart';
+import 'package:devtools_app/src/screens/memory/panes/allocation_tracing/allocation_profile_tracing_tree.dart';
+import 'package:devtools_app/src/screens/memory/panes/allocation_tracing/allocation_profile_tracing_view.dart';
+import 'package:devtools_app/src/screens/memory/panes/allocation_tracing/allocation_profile_tracing_view_controller.dart';
+import 'package:devtools_app/src/shared/notifications.dart';
+import 'package:devtools_shared/devtools_shared.dart';
+import 'package:devtools_test/devtools_test.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../test_data/memory_allocation.dart';
+
+void main() {
+  late FakeServiceManager fakeServiceManager;
+
+  void _setUpServiceManager() {
+    // Load canned data testHeapSampleData.
+    final allocationJson =
+        AllocationMemoryJson.decode(argJsonString: testAllocationData);
+
+    final classList = ClassList(
+      classes: [
+        ClassRef(id: 'cls/1', name: 'ClassA'),
+        ClassRef(id: 'cls/2', name: 'ClassB'),
+        ClassRef(id: 'cls/3', name: 'ClassC'),
+      ],
+    );
+
+    fakeServiceManager = FakeServiceManager(
+      service: FakeServiceManager.createFakeService(
+        allocationData: allocationJson,
+        classList: classList,
+      ),
+    );
+    mockConnectedApp(
+      fakeServiceManager.connectedApp!,
+      isFlutterApp: true,
+      isProfileBuild: false,
+      isWebApp: false,
+    );
+    setGlobal(ServiceConnectionManager, fakeServiceManager);
+  }
+
+  Future<void> pumpMemoryScreen(WidgetTester tester) async {
+    await tester.pumpWidget(
+      wrapWithControllers(
+        const MemoryBody(),
+        memory: MemoryController(),
+      ),
+    );
+
+    // Delay to ensure the memory profiler has collected data.
+    await tester.pumpAndSettle(const Duration(seconds: 1));
+    expect(find.byType(MemoryBody), findsOneWidget);
+  }
+
+  // Set a wide enough screen width that we do not run into overflow.
+  const windowSize = Size(2225.0, 1000.0);
+  setGlobal(NotificationService, NotificationService());
```

Comment:
is there a reason this is called outside of `setUp`?

Reply:
Because it was in `memory_screen_test.dart` and I copy-pasted the original test... :-)